### DISCUSSION TITLE
[Support] midway 框架 中 swagger 模块无法在 windows7 环境下 IE 浏览器中打开

### DISCUSSION THREAD

**Discussion by zhuzeyu22** [midwayjs/midway#3506] [Q&A]
### Describe the problem（描述问题）

midway 框架 中 swagger 模块无法在 windows7 环境下 IE 浏览器中打开
报错信息  swagger-ui.bundle.js 2:252 语法错误

### Midway Versions（Midway 版本）

3.9.0

**Comment by waitingsong** [comment]
应该是 win7 IE版本太老（IE9？）
可以升级到 IE11 或者安装 firefox、chrome

**Reply by zhuzeyu22** [reply]
需要适配 win7 环境以及 IE 浏览器，我尝试将 tsconfig target 设置成 ES5 但是编译不通过，请问我应该在哪里修改编译配置（比如 babel ？）

**Reply by czy88840616** [reply]
你可能需要自己编译一个 https://www.npmjs.com/package/swagger-ui-dist ？

**Reply by zhuzeyu22** [reply]
这多少有点麻烦，如果有工具可以直接编译或者加一些 patch 就好了

**Reply by czy88840616** [reply]
这个包是swagger 官方提供的，我们无能为力啊。

**Reply by zhuzeyu22** [reply]
如果我想要修改 midway 项目的 webpack 和 babel 配置我应该怎么做？  
btw 如果我阅读 midway 的源码用 yarn 做包管理可以吗？还是说我需要学习下 lerna ？

**Reply by zhuzeyu22** [reply]
环境方面的文档资料太少了，封装的比较封闭。

**Reply by czy88840616** [reply]
环境? Midway 只用了 typescript 的 tsc 将 ts 转换为 js，没有用别的工具。阅读源码不需要学习工具，只需要 npm i , npm run build 即可（没有用 yarn）。
